### DISCUSSION TITLE
Alternative for chai-like and chai-things in chai v5 for partial fuzzy matching?

### DISCUSSION THREAD

**Discussion by melink14** [chaijs/chai#1634] [Support]
I have been using chai-things and chai-like to get fuzzy partial matches of arrays containing objects. Both of these don't look like they'll get updates for v5 anytime soon.

```ts
expect(
  rcxDict.wordSearch('ＤＶＤ-ＲＯＭ', /* doNames= */ false)?.data
).to.include.something.like({ entry: /^ＤＶＤ-ＲＯＭ .*/ });
```

Here the acutal value is an object with various keys and we're testing that one of the keys has a string which matches a regex.

```ts
expect(rcxDict.deinflect('です')).to.include.something.like({
  word: 'だ',
});
```

Here, we're getting an array of objects with various keys and one of the keys should match. (in this case regex wasn't necessary)

I saw https://github.com/chaijs/chai/issues/644 mentioned chai v5 having loose matchers for the deep assertions but wasn't able to get something like that to work.

Thanks.

**Comment by keithamus** [comment]
We didn't get around to loose matchers in v5. It'll likely be in a follow up release.

**Reply by melink14** [reply]
Is the best alternative then the sinon matcher inline plugin? (or I guess petitioning the plugin owners (or forking them))

thanks for hte answer!

**Reply by keithamus** [reply]
I don't know what the best is. We laid the groundwork for fuzzy matchers; `deep-eql` can have a custom comparator. So if someone wanted to write a plugin (or even better a PR to core!) based on the ground work we did, I'm sure we could make it happen in v5.x.
